compiler/rustc_const_eval/src/interpret/machine.rs

```diff
@@ -36,6 +36,9 @@ pub enum StackPopJump {
     /// Indicates that we should *not* jump to the return/unwind address, as the callback already
     /// took care of everything.
     NoJump,
+
+    /// Returned by [`InterpCx::pop_stack_frame`] when no cleanup should be done.
+    NoCleanup,
```

Comment:
But what does it mean if a machine hook returns this?

Reply:
Hm. My answer would probably be "it shouldn't return this", but I have no idea on how to enforce this...

Reply:
As a start, an assertion would be an option?

Reply:
Assertion added in [54e11cf](https://github.com/rust-lang/rust/pull/113128/commits/54e11cf378d5f3a423d98b9a8bcbc6c25078995a)